MovieNight/MovieNightApiClient.swift

```diff
@@ -0,0 +1,137 @@
+//
+//  SWApiCllient.swift
+//  TheApiAwakens
+//
+//  Created by Tassia Serrao on 19/01/2017.
+//  Copyright © 2017 Tassia Serrao. All rights reserved.
+//
+
+import Foundation
+
+protocol Endpoint {
+    var baseURL: String { get }
+    var path: String { get }
+    var parameters: [String: String] { get }
+}
+
+extension Endpoint {
+    var queryItems: [URLQueryItem] {
+        var queryItems = [URLQueryItem]()
+        
+        for (key, value) in parameters {
+                let queryItem = URLQueryItem(name: key, value: "\(value)")
+                queryItems.append(queryItem)
+        }
+        return queryItems
+    }
+    
+    var request: URLRequest {
+        let components = NSURLComponents(string: baseURL)!
+        components.path = path
+        components.queryItems = queryItems // the URL is percent encoded here
+        
+        let url = components.url!
+        return URLRequest(url: url)
+    }
+}
+
+enum MovieNightEndpoint: Endpoint {
+    case Genre
+    case Actor(page: String)
+    case Movie(page: String)
+    
+    var baseURL: String {
+        return "https://api.themoviedb.org"
+    }
+    var path: String {
+        switch self {
+            case .Genre:
+                return "/3/genre/movie/list"
+            case .Actor:
+                return "/3/person/popular"
+            case .Movie:
+                return "/3/movie/popular"
+        }
+    }
+    
+    var parameters: [String : String] {
+        var parameters = [String : String]()
+        switch self {
+        case .Actor(let page), .Movie(let page) :
+            parameters["page"] = page
+            parameters["api_key"] = "57b213eb9d700e45c3f1ddaa754d7134"
```

Comment:
I changed the logic a bit. Although GetGenre endpoint had only one page at the moment I'm allowing it to receive pages as a parameter as well in case they increase the genres' list - who knows..